client/src/pages/DetailedWorkoutPage.jsx

```diff
@@ -1,19 +1,51 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
 const DetailedWorkoutPage = () => {
-  const workout = {
-    //able to add more as needed
-    title: "Sample Workout",
-    videoUrl: "https://www.youtube.com/embed/ErZhd3qvPYs",
-    instructions: "Go Brazy For five sets of six reps",
-    type: "Cardio",
-    muscle: "Legs",
-    difficulty: "Intermediate",
+
+  const { name } = useParams();
+  const [ workout, setWorkout ] = useState(null);
+  const [ error, setError ] = useState(null);
+  const [ showFullInstructions, setShowFullInstructions ] = useState(false);
+
+  useEffect(() => {
+
+    const getWorkout = async () => {
+      try {
+        const response = await axios.get(`http://localhost:3001/api/workouts/${name}`); 
+        setWorkout(response.data);
+      } catch (error) {
+        setError(error.response.data.error);
+      }
+    };
+
+    getWorkout();
+
+  }, [name]);
```

Comment:
- While useEffect() is possible, I recommend using RTK queries instead. 
- Take a look at the `WorkoutServices` on the redux folder. We have the same endpoint defined. 
- This is better for performance since it caches the result for future use
- It also good for code reusability